backend/app/api/schedules.py

```
@@ -0,0 +1,207 @@
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Error getting schedule: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to get schedule: {str(e)}")
+
+
+@router.put("/{schedule_id}", response_model=ScheduleResponse)
+async def update_schedule(
+    schedule_id: int,
+    schedule_update: dict,
+    db: Session = Depends(get_db)
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)